lunaria/components.ts

```diff
@@ -0,0 +1,75 @@
+import { html } from '@lunariajs/core';
+import { githubGet } from './github-get.mjs';
+
+export const TitleParagraph = () =>
+  html`<p>
+      If you're interested in helping us translate
+      <a href="https://v2.tauri.app/">v2.tauri.app</a> into one of the languages listed below,
+      you've come to the right place! This auto-updating page always lists all the content that
+      could use your help right now.
+    </p>
+
+    <p>
+      Read the
+      <a href="https://github.com/tauri-apps/tauri-docs/blob/v2/.github/TRANSLATING.md"
+        >Translations Guide</a
+      >
+      for how to translate a document. Before starting a new translation, be sure to check the
+      <a href="#needs-review">existing Tauri Docs PRs</a> to see if this page has already been
+      translated, and consider reviewing any open PRs in your language!
+    </p>`;
+
+interface PullRequest {
+  html_url: string;
+  title: string;
+  labels: {
+    name: string;
+  }[];
+}
+interface Options {
+  githubRepo: string;
+  githubToken: string;
+}
+async function getPullRequests(options: Options) {
+  const pullRequests: PullRequest[] = await githubGet({
+    url: `https://api.github.com/repos/${options.githubRepo}/pulls?state=open&per_page=100`,
+    githubToken: options.githubToken,
```

Comment:
```suggestion

```
would you test removing the token altogether?  I'm pretty sure it should work without it. If not I think we need to setup a pat with read-only on pull requests

https://api.github.com/repos/tauri-apps/tauri-docs/pulls?state=open&per_page=100

https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-pull-requests

Reply:
even if it works, leave it commented out, not sure if we're gonna get rate limited or something

Reply:
oh, and I'm not so sure but this function would need to be async, maybe we can get away from it if we create a small script that will fetch the data on `pnpm dev:setup`  and create a static md with the necessary snippet, then when running `dev` or `build` it would be readily available, I haven't used Lunaria so I'm not sure this will be needed.

Reply:
> oh, and I'm not so sure but this function would need to be async, maybe we can get away from it if we create a small script that will fetch the data on `pnpm dev:setup` and create a static md with the necessary snippet, then when running `dev` or `build` it would be readily available, I haven't used Lunaria so I'm not sure this will be needed.

The "Translations that need reviews" module needs to be updated in real time so you can't use static md

Reply:
Asynchrony in js is difficult to eliminate, and slots in Lunaria must not be asynchronous functions.

Reply:
> but you'll need to update the page every time there's a new i18n-related PR.


Ah, you're right, I forgot about that detail

Reply:
what if we make an webcomponent? It should be able to render sync and once the promise resolve, update the content.

Reply:
ughhh, this might not so straight forward as I'd thought... Do you have any ideas?

Reply:
> https://api.github.com/repos/tauri-apps/tauri-docs/pulls?state=open&per_page=100

`https://api.github.com/repos/tauri-apps/tauri-docs/pulls?state=open&per_page=100`

I tried to put the api in the html, but the api was throwing errors because of cross-origin issues.
I recommend dropping the "[Translations that need reviews](https://tauri.app/contribute/translate-status#needs-review)" module

Reply:
> ughhh, this might not so straight forward as I'd thought... Do you have any ideas?

`https://api.github.com/repos/tauri-apps/tauri-docs/pulls?state=open&per_page=100`

I tried to put the api in the html, but the api was throwing errors because of cross-origin issues.
I recommend dropping the "[Translations that need reviews](https://tauri.app/contribute/translate-status#needs-review)" module

Reply:
we can drop it fow now and create a new issue so we can track

Reply:
>I tried to put the api in the html, but the api was throwing errors because of cross-origin issues.

how come?

Reply:
> we can drop it fow now and create a new issue so we can track

Agree, i will remove `Translations that need reviews` module.

Reply:
> how come?

![image](https://github.com/user-attachments/assets/fb9dcec3-14d8-4d9e-a4e8-b079769d5718)
There should be a restriction on csp in the documentation.

Reply:
ah indeed there's a headers file, I forgot about that. Assuming we're removing `Translations that need reviews` we don't need to update the headers right? If that's the only the issue then we probably can work around it. https://github.com/tauri-apps/tauri-docs/blob/v2/public/_headers

Reply:
> ah indeed there's a headers file, I forgot about that. Assuming we're removing `Translations that need reviews` we don't need to update the headers right? If that's the only the issue then we probably can work around it. https://github.com/tauri-apps/tauri-docs/blob/v2/public/_headers

yes